std/src/main/scala/algebra/std/list.scala

```diff
@@ -11,15 +12,21 @@ trait ListInstances {
 }
 
 class ListOrder[A](implicit ev: Order[A]) extends Order[List[A]] {
-  def compare(x: List[A], y: List[A]): Int =
-    (x, y) match {
-      case (Nil, Nil) => 0
-      case (Nil, _) => -1
-      case (_, Nil) => 1
-      case (x :: xs, y :: ys) =>
-        val n = ev.compare(x, y)
-        if (n != 0) n else compare(xs, ys)
-    }
+  def compare(xs: List[A], ys: List[A]): Int = {
+    @tailrec def loop(xs: List[A], ys: List[A]): Int =
+      xs match {
+        case Nil =>
```

Comment:
Just realised that @non optimized the emptyListLeft case a bit - so updating benchmark to add this implementation.